index.ts

```diff
@@ -20,38 +20,60 @@ export interface EntryInfo {
   basename: string;
 }
 export type PathOrDirent = Dirent | Path;
-export type Tester = (path: EntryInfo) => boolean;
+export type Tester = (entryInfo: EntryInfo) => boolean;
 export type Predicate = string[] | string | Tester;
-
-function defaultOptions() {
-  return {
-    root: '.',
-    fileFilter: (_path: EntryInfo) => true,
-    directoryFilter: (_path: EntryInfo) => true,
-    type: FILE_TYPE,
-    lstat: false,
-    depth: 2147483648,
-    alwaysStat: false,
-    highWaterMark: 4096,
-  };
+export const enum EntryTypes {
+  FILE_TYPE = 'files',
+  DIR_TYPE = 'directories',
+  FILE_DIR_TYPE = 'files_directories',
+  EVERYTHING_TYPE = 'all',
 }
+export type ReaddirpOptions = {
+  root: string;
+  fileFilter?: Predicate;
+  directoryFilter?: Predicate;
+  type?: EntryTypes | 'files' | 'directories' | 'files_directories' | 'all';
```

Comment:
another duplication

Reply:
but then you have the breaking change... it then required you to pass an enum value and not the string value.

Reply:
thanks for clarification